spec/lib/open_food_network/reports/base_report_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+
+module OpenFoodNetwork::Reports
+  describe BaseReport do
+    context 'inheritance methods to search and filter for ui-grid reports' do
+      let(:user) { create(:admin_user) }
+
+      context 'blank object' do
+        let(:report) { BaseReport.new(user, {query: 'keyword'}) }
+        it 'new instance accepts 3 arguments' do
+          expect(report.params).to eq({query: 'keyword'})
+        end
+
+        it 'gets search object via permissions' do
+          expect(report.search).to be_instance_of(Ransack::Search)
+        end
+
+        it 'returns empty array of line_items' do
+          expect(report.table_items).to be_empty
+        end
+      end
+
+      context 'with line items' do
+        let(:enterprise) { create(:distributor_enterprise) }
+        let(:order_cycle) { create(:simple_order_cycle) }
+        let(:order) { create(:order, completed_at: 1.day.ago, order_cycle: order_cycle, distributor: enterprise) }
+        let(:line_item) { build(:line_item) }
+        let(:report) { BaseReport.new(user) }
+
+        before { order.line_items << line_item }
+
+        it '#table_items returns one line_item' do
+          expect(report.table_items).to eq([line_item])
```

Comment:
I would normally do `.to eq [line_item]`. Do you have a preference for doing it with parentheses? Doesn't really matter, I'm more thinking about what we do as a convention.